cfme/tests/cloud_infra_common/test_cockpit_server.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+import fauxfactory
+
+from cfme.cloud.instance import Instance
+from cfme.cloud.provider import CloudProvider
+from cfme.cloud.provider.ec2 import EC2Provider
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.common.provider import CloudInfraProvider
+from cfme.infrastructure.virtual_machines import Vm
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.version import current_version
+
+
+@pytest.fixture(scope="function")
+def new_vm(provider, request):
+
+    if isinstance(provider, CloudProvider):
+        vm = Instance(name="test_cockpit{}".format(fauxfactory.gen_alphanumeric()),
+                      provider=provider)
+    else:
+        vm = Vm(name="test_cockpit{}".format(fauxfactory.gen_alphanumeric()), provider=provider)
+    if not provider.mgmt.does_vm_exist(vm.name):
+        vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+        request.addfinalizer(vm.delete_from_provider)
+    return vm
+
+
+@pytest.mark.uncollectif(
+    lambda provider: current_version() < "5.9" or provider.type == 'gce')
+@pytest.mark.provider([CloudInfraProvider])
+@pytest.mark.parametrize('enable', [False, True])
+def test_cockpit_server_role(appliance, provider, setup_provider, new_vm, enable):
+    """
+    The test checks the cockpit "Web Console" button enable and disabled working.
+    """
+
+    if enable:
+        appliance.server.settings.enable_server_roles('cockpit_ws')
+    view = navigate_to(new_vm, 'Details')
+    if enable:
+        if provider.one_of(AzureProvider, EC2Provider):
```

Comment:
please put a comment about what happens to web_console button when providers are AzureProvider, EC2Provider